lib/src/facts/aix/kernel_resolver.cc

```diff
@@ -0,0 +1,70 @@
+#include <internal/facts/aix/kernel_resolver.hpp>
+#include <internal/util/regex.hpp>
+#include <facter/facts/collection.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <facter/execution/execution.hpp>
+#include <facter/util/file.hpp>
+#include <boost/regex.hpp>
+
+using namespace facter::util;
+using namespace facter::execution;
+using namespace std;
+
+namespace facter { namespace facts { namespace aix   {
+
+    kernel_resolver::data kernel_resolver::collect_data(collection& facts)
+    {
+        data result;
+        const auto regex = boost::regex("(\\d+-\\d+-\\d+)_SP.*Service Pack");
+        vector<string> versions;
+
+        // We reimplement part of oslevel here. We parse the rml cache
+        // and use that to determine the full AIX system version. If
+        // the rml cache doesn't exist, we just call out to oslevel to
+        // regenerate it (since we're gonna be slow in that case
+        // anyway).
+        bool success;
+        bool tried_oslevel = false;
+        do {
```

Comment:
This is just 3 steps:
* Try to read .oslevel.rml.cache
* If that fails, run oslevel
* Try reading .oslevel.rml.cache again

It seems simpler to just have a function that reads .oslevel.rml.cache and just have the 3 steps explicitly. The loop complicates things.